blocks/blogsearch/blogsearch.js

```diff
@@ -1,15 +1,69 @@
 import {
-  div, form, input, i,
+  a, div, form, input, i,
 } from '../../scripts/dom-helpers.js';
+import { getLocaleInfo, debounce } from '../../scripts/scripts.js';
+import ffetch from '../../scripts/ffetch.js';
+
+async function handleSearch(block) {
+  const searchValue = block.querySelector('input').value;
+  if (searchValue.length < 3) {
+    return;
+  }
+
+  const searchResults = block.querySelector('.search-results');
+  searchResults.innerHTML = '';
+
+  console.log(`searching for ${searchValue}`);
+
+  const entries = await ffetch(`${getLocaleInfo().metadataIndex}`)
+    .sheet('blogs-content')
+    .filter((entry) => {
+      const searchTerms = searchValue.toLowerCase().split(/\s+/);
+      const prefixLength = getLocaleInfo().placeholdersPrefix.length;
+
+      return searchTerms.some((term) => (entry.title && entry.title.toLowerCase()
+        .includes(term))
+          || (entry.description && entry.description.toLowerCase()
+            .includes(term))
+          || (entry.header && entry.header.toLowerCase()
+            .includes(term))
+        || (entry.path && entry.path.substring(prefixLength).toLowerCase()
+          .includes(term))
+          || (entry.content && entry.content.toLowerCase()
+            .includes(term)),
+      );
+    })
+    .all();
+
+  // potential improvement: search content separately, first show those
+  // that match on title, description and header
+  //
+  // while sorting by publicationDate would potentially make sense, it's not what's currently
+  // implemented on servicenow.com
```

Comment:
The original implementation seems to indeed split the results into 2 categories:
1. Results where the search term appears directly in the header/title.
2. Results where the search term appears in the content of the blog

The results from 1. appear first in the list with the search term highlighted.
The results from 2. appear afterwards in the list.

Reply:
OK, I'll add that to the implementation then.